plugins.md

```diff
@@ -0,0 +1,218 @@
+
+A shortcut for calling `api.getTransaction()` followed by
+`transaction.runInChildSpan(options, fn)`. Note that while this function accepts
+the same arguments as `transaction.runInChildSpan`, the function being passed in
+should handle being passed `null` as a parameter. This will be the case if
+`api.getTransaction` returns `null`.
+
+#### Trace Span Options
+
+Some functions above accept a `TraceOptions` object, which has the following
+fields:
+
+* `name`: `string`
+  * Required
+  * The name that should be given to the newly created span.
+* `traceContext`: `string`
+  * Optional for root spans, ignored for child spans
+  * A serialized trace context. If the module being traced is a web framework,
+    the plugin that patches this module should attempt to extract this from
+    an incoming request header and set this field. Omitting this field will
+    never cause an error, but may cause trace spans that correspond to a single
+    request across several services to appear disassociated with each other.
+    See also [Cross-Service Trace Contexts](#cross-service-trace-contexts).
+* `url`: `string`
+  * Optional for root spans, ignored for child spans
+  * The URL of the incoming request. This only applies if the module being
+    traced is a web framework. If given, this field will be compared against the
+    trace agent's URL filtering policy to check whether a span should be
+    created.
+* `skipFrames`: `number`
+  * Optional; defaults to `0`
+  * Trace spans include the call stack at the moment of creation as part of the
+    information gathered. The call stack may include undesirable frames such as
+    frames within the plugin itself. This field specifies the number of stack
+    frames to skip when writing the call stack to the trace span. Frames within
+    the trace agent implementation are automatically skipped.
+
+### Trace Agent Configuration
+
+#### `api.enhancedReportingEnabled()`
+* Returns `boolean`
+
+Returns a boolean value describing whether the trace agent was started with
+an enhanced level of reporting. See the [configuration][config-js] object
+definition for more details.
+
+### Cross-Service Trace Contexts
+
+The Trace Agent supports distributed tracing, so that in supported web
+frameworks, incoming requests that are known to come from other services that
+are also integrated with Stackdriver Trace (through the
+['x-cloud-trace-context'][stackdriver-trace-faq]
+field in request headers) should build spans that are aware of the information
+serialized in this field, known as the trace context.
+
+It is up to plugin developers to extract serialized trace context from incoming
+requests and propagate it in outgoing requests. The Plugin API accepts the
+serialized trace context as an [option](#trace-span-options) when creating new
+trace spans.
+
+#### `transaction.getTraceContext()` | `childSpan.getTraceContext()`
+* Returns `string`
+
+Gets the trace context serialized as a string.
+
+### Context Propagation
```

Comment:
It would be good to include a little bit here on context propagation (like a sentence or two) even for people familiar with CLS.